src/fileMapper.ts

```diff
@@ -389,20 +385,27 @@ export class FileMapper implements Disposable {
 
     /**
      * Tries to remapped path using any of the RootPaths in the config
-     * @param uri file uri to try to rebase
      */
-    private tryConfigRootPathsUri(uri: Uri, uriBase?: string): Uri | undefined {
-        const originPath: path.ParsedPath = path.parse(Utilities.getFsPathWithFragment(uri));
-        const dir: string = originPath.dir.replace(originPath.root, '');
-
-        for (const rootpath of this.rootPaths) {
-            const dirParts: string[] = dir.split(path.sep);
-            dirParts.push(originPath.base);
-
+    private tryConfigRootPathsUri(originUri: Uri, uriBase?: string, rootPaths?: Uri[]): Uri | undefined {
+        // Parse the remote URI into directory parts.
+        const originPath: path.ParsedPath = path.parse(originUri.fsPath);
+
+        // Remove the "root" of the path. (For example, if the incoming path is e:\, remove it.)
+        // So given an incoming URI such as "e:\foo\bar\xyz.cpp" we are left with "\foo\bar"
+        const originPathWithoutRoot: string = path.join(originPath.dir.replace(originPath.root, ''), originPath.base);
```

Comment:
>originPath.dir.replace(originPath.root, '') [](start = 56, length = 43)

just making sure - does this work as expected on linux/mac?

Reply:
Yep.

---
In reply to: [418157101](https://github.com/microsoft/sarif-vscode-extension/pull/276#discussion_r418157101) [](ancestors = 418157101)